### DISCUSSION TITLE
Can Postfix be on a cloud instance?

### DISCUSSION THREAD

**Discussion by whisperdancer** [simple-login/app#1362] [Self-hosting question]
I'm running my simplelogin in a cloudflared tunnel which I now discover blocks port 25 so I do not receive my verification email and never will. I'm curious if it's possible to keep my simplelogin running where it is, but move my postfix to an AWS instance running in the cloud? If so, how would I connect my simplelogin to the postfix running in the cloud?

Thank you!

**Comment by nguyenkims** [comment]
@whisperdancer you can have Postfix sending the email to the email handler container on a distant host by changing the `smtp:127.0.0.1:20381` to `smtp:{sl_server_ip_address}:20381`. Make sure that the port 20381 isn't blocked by a firewall or ISP. It's also better to set up a firewall rule so only the EC2 instance can reach out to this port to avoid being attacked.

**Comment by whisperdancer** [comment]
Thank you @nguyenkims I appreciate your quick response! Just to clarify that I understand correctly. I would only make the changes to the query in /etc/postfix/pgsql-transport-maps.cf like this?

```
query = SELECT 'smtp:{sl_server_ip_address}:20381' FROM custom_domain WHERE domain = '%s' AND verified=true
    UNION SELECT 'smtp:{sl_server_ip_address}:20381' FROM public_domain WHERE domain = '%s'
    UNION SELECT 'smtp:{sl_server_ip_address}:20381' WHERE '%s' = 'mydomain.com' LIMIT 1;
```
    
No other configuration changes would be necessary?

Thank you kindly!

**Reply by nguyenkims** [reply]
Yes I think this change should be enough. We don't use this technique but I think it should work.
